test/libsolidity/semanticTests/inlineAssembly/transient_storage_simple_reentrancy_lock.sol

```diff
@@ -0,0 +1,22 @@
+contract C {
+  modifier nonreentrant {
+    assembly {
+      if tload(0) { revert(0, 0) }
+      tstore(0, 1)
+    }
+    _;
+   assembly {
+      tstore(0, 0)
+   }
+  }
+  function f(bool simulateReentrancy) nonreentrant public {
+    if (simulateReentrancy) {
+       f(false);
+    }
+  }
+}
+// ====
+// EVMVersion: >=cancun
+// ----
+// f(bool): false -> FAILURE
```

Comment:
This should not fail, though :-).
If it does, we need to check what happens and why exactly.

Reply:
Might be easier to start with more and simpler tests then. Just one store. Storing and loading back.
In general, we're also indepedently missing further tests like: it's valid to load, but not store, in ``staticcall`` context (resp. external view functions)

Reply:
Isn't it because evmone with cancun support was not released yet? So we still using version [0.10.0](https://github.com/ethereum/evmone/releases/tag/v0.10.0). See also: https://github.com/ethereum/solidity/pull/14705#issuecomment-1832509938

Reply:
At some point you said that we need to upgrade, but then @nikola-matic said evmone support should already be there :-D. But it looks very much like it may not be. If they don't yet have a release that's a bit of a hassle, though...

Reply:
Weidly https://github.com/ethereum/evmc/releases/tag/v11.0.0 claims cancun support, it's a bit confusing.

Reply:
I just talked to Pawel, we'll get an evmone release tomorrow.

Reply:
@matheusaaguiar you can either change the base branch to https://github.com/ethereum/solidity/pull/14743, which should allow you to work with evmone 11 locally (you'll need to upgrade the .so in solidity/deps), or you can just cherry pick the first two commits locally, implement the functionality in `EVMHost.cpp` (if needed), and then just drop them before pushing (that's what I did for `blobsafefee`).

Reply:
evmone update is already on `develop` and I rebased the PR on that, so now it should just work after you update the expectations.